app/server/lib/AttachmentStoreProvider.ts

```diff
@@ -173,10 +173,10 @@ export async function getConfiguredAttachmentStoreConfigs(): Promise<IAttachment
     const snapshotProvider = create.getAttachmentStoreOptions().snapshots;
     // This shouldn't happen - it could only happen if a version of Grist removes the snapshot provider from ICreate.
     if (snapshotProvider === undefined) {
-      throw new Error("Snapshot provider is not available on this version of Grist");
+      return [];
```

Comment:
Not here, but somewhere it feels like a log message along the lines of "no snapshot storage configuration found, external attachments also disabled" would be useful.

Reply:
I agree, but couldn't think where the most sensible place would be - it's kind of here?

This is where the config value is translated into a usable object for the rest of the system, and validity checks happen. It's pretty much the only part of the system that's aware of "User has requested snapshots, snapshots not available".

We could, effectively, memoize this function, so it only fires a log entry the first time it's called?